src/snyk/base/services/authenticationService.ts

```diff
@@ -107,6 +108,17 @@ export class AuthenticationService implements IAuthenticationService {
     await this.configuration.setToken(token);
   }
 
+  async updateToken(token: string): Promise<void> {
+    if (!token) {
+      await this.initiateLogout();
+    } else {
+      if (!uuidValidate(token)) return Promise.reject(new Error('The entered token has an invalid format.'));
+
+      await this.initiateLogout();
+      await this.configuration.setToken(token);
```

Comment:
You have to handle context here, similarly to `snykLib.ts` smth like:
```ts
      await this.contextService.setContext(SNYK_CONTEXT.AUTHENTICATING, false);
      await this.contextService.setContext(SNYK_CONTEXT.LOGGEDIN, true);
```
Also you have to disable loading badge on the extension icon in the side bar by calling:
`this.loadingBadge.setLoadingBadge(false);`

Search for it in the code to see how it's used.

It's kinda hard to know it because you can't test it well. Maybe you can try @asaf92's LSP tool to send a custom message to visually test how the extension will react to it?

Reply:
Oh, another easier way to test it is to enable one of the products temporarily in code (`initializationOptions`) when spawning LS and open an editor. This initiate authentication process and then you can see how your logic behaves :)

Reply:
👍